frontend/src/components/UserTables/LocationsTable.tsx

```diff
@@ -0,0 +1,85 @@
+import React, { useState, useEffect } from 'react';
+import Form from '../UserForms/LocationsForm';
+import { Location } from '../../types';
+import './table.css';
+
+const Table = () => {
+  const [locations, setLocations] = useState<Location[]>([]);
+
+  const getExistingLocations = async () => {
+    const locationsData = await fetch('/locations')
+      .then((res) => res.json())
+      .then((data) => data.data);
+    setLocations(locationsData.map((location: any) => ({
+      id: location.id,
+      name: location.name,
+      address: location.address,
+    })));
+  };
+
+  useEffect(() => {
+    getExistingLocations();
+  }, []);
```

Comment:
Nice. Because otherwise, the setLocations call in getExistingLocations would've given an infinite loop.